### DISCUSSION TITLE
iOS Tracking Authorization and IDFA

### DISCUSSION THREAD

**Discussion by nijakobius** [capawesome-team/capacitor-firebase#309] [Q&A] [answered]
Could you please offer some guidance on the implications these plugins have on privacy and Apple's Tracking Authorization prompt? :)

1. If the iOS user selects "Ask App Not to Track", I guess IDFA collection needs to be disabled?

2. But can this be done after app launch?

3. Here it appears the decision over IDFA collection must be made even before compiling the app: https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/analytics#disable-idfa-collection

4. In 1. - 3. I was only talking about the Analytics plugin, but are there aspects of other plugins in this bundle that need to be limited/disabled when the user opts out of tracking?

**Comment by robingenz** [comment, marked as answer]
The best thing to do is to read the privacy statements from Firebase itself. I only use the SDKs from Firebase (with a few exceptions, e.g. [`third-party-sdks.md`](https://github.com/capawesome-team/capacitor-firebase/blob/main/packages/authentication/docs/third-party-sdks.md)). The Capacitor plugins themselves do not collect any data, only the SDKs which are used.

1. It is up to you how you implement it. I disable IDFA collection by default in my apps and only enable it when the user has agreed (see [Firebase Docs](https://firebase.google.com/docs/analytics/configure-data-collection?platform=ios#temporarily_disable_collection_2)).
2. Depends on the plugin. Capacitor Firebase Analytics, for example, has a [`setEnabled(...)`](https://github.com/capawesome-team/capacitor-firebase/tree/main/packages/analytics#setenabled) method. 
3. Yes, if you are sure that you do not want to use `IDFA collection` at all, then you can remove it completely.
4. Please look directly at the Firebase documentation. If I make a statement here, it could be outdated tomorrow.
